modules/index.js

```diff
@@ -1,14 +1,30 @@
 const payloadPassThrough = (state, payload) => payload;
 
+const handlePayloadPassThrough = (actionSpec) => {
+  const maybeActionReducer = actionSpec.slice(-1)[0];
+  const allButLast = actionSpec.slice(0, -1);
+
+  if (typeof maybeActionReducer === 'function') {
+    return { actionTypes: allButLast, actionReducer: maybeActionReducer };
+  }
+  if (typeof maybeActionReducer === 'string') {
+    return { actionTypes: actionSpec, actionReducer: payloadPassThrough };
+  }
+  throw new Error('Reducer must either be a function or not present (payloadPassThrough)');
+}
+
 const createReducer = (...actionHandlers) => (defaultValue = null) => {
     const actions = actionHandlers.reduce(
         (acc, actionSpec) => {
             actionSpec = [].concat(actionSpec);
-            const last = actionSpec.slice(-1)[0];
-            const actionReducer = typeof last === 'function' ? last : payloadPassThrough;
-            const actionTypes = actionReducer === payloadPassThrough ? actionSpec : actionSpec.slice(0, -1);
-
-            actionTypes.forEach(actionType => acc[actionType] = actionReducer);
+            const { actionTypes, actionReducer } = handlePayloadPassThrough(actionSpec)
+
+            actionTypes.forEach(actionType => {
+              if (typeof actionType === 'undefined') {
+                throw new Error('actionTypes cannot be undefined')
```

Comment:
`'[redux-action-reducer] An action type supplied to `createReducer` is undefined'`